elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/BulkRequest.java

```diff
@@ -78,6 +83,45 @@ public void execute() {
 		}
 	}
 
+	/**
+	 * Call every result handlers, managing exceptions so that one failing handler will not prevent
+	 * others from being called.
+	 *
+	 * <p>No locally caught exception will be re-thrown, but the "main" exception will be returned.
+	 *
+	 * <p>The "main" exception is either the {@code preexistingMainException} parameter (if non-null) or the first
+	 * locally caught exception. Every locally caught exception that is not the "main" exception will be
+	 * added to the "main" exception.
+	 *
+	 * @param successfulItems The requests and their results, for calling result handler.
+	 * @param preexistingMainException The exception to use as a "main" exception (see above), or {@code null}.
+	 * @return The "main" exception (see above), or {@code null} if there isn't any.
+	 */
+	private RuntimeException callHandlers(Map<BackendRequest<?>, BulkResultItem> successfulItems, RuntimeException preexistingMainException) {
+		RuntimeException mainException = preexistingMainException;
+		for ( Map.Entry<BackendRequest<?>, BulkResultItem> entry : successfulItems.entrySet() ) {
```

Comment:
I'll see what I can do...
But I'd like to point out that the only other uses of `IndexingMonitor` only increment it by 1. Which is why I implemented it that way: it sure didn't seem to be a concern with the Lucene indexing service, so I figured there was not reason it should be a concern with the Elasticsearch indexing service. Maybe we should have a look at the Lucene side too (`org.hibernate.search.backend.impl.lucene.works.AddWorkExecutor.performWork(LuceneWork, IndexWriterDelegate, IndexingMonitor)` for instance).